packages/ui-extensions/src/surfaces/point-of-sale/run/input/BaseInput.ts

```diff
@@ -0,0 +1,13 @@
+import type {
+  ConnectivityState,
+  DeviceApiContent,
+  Session,
+} from '../../../point-of-sale';
+import type {CurrencyCode} from '../../../../shared';
+
+export interface BaseInput {
+  connectivity: ConnectivityState;
+  device: DeviceApiContent;
```

Comment:
DeviceApiContent has a bunch of getter functions which are very "API-like". I think we should re-write this into an interface that just has read values, and POS can call the functions on its side and return the values there.

Reply:
added.

Reply:
In the near future, different locations can have the locale/currency which may differ than what is set in the store's admin.